docker/build_matrix.yaml

```diff
@@ -43,97 +43,116 @@
   PYTORCH_VERSION: 1.12.1
   TAGS:
   - mosaicml/pytorch:1.12.1_cu116-python3.9-ubuntu20.04
-  - mosaicml/pytorch:latest
```

Comment:
I'm thinking we don't move the `latest` tag just yet since the focus of this PR is to just start building the Torch 1.13 images.  I think Docker convention is to typically point `latest` to the latest _stable_ image and even though we don't explicitly version these image just yet, the Torch 1.12 image is still technically the most stable.  We can update the `latest` pointer to Torch 1.13 after we have some soak time.  I also don't think we want to accidentally force customers to update to Torch 1.13 if they're using `latest`.